platforms/web/lib/useWysiwyg.formatting.test.tsx

```diff
@@ -0,0 +1,184 @@
+/*
+Copyright 2022 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import {
+    act,
+    fireEvent,
+    render,
+    screen,
+    waitFor,
+} from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+
+import { Editor } from './testUtils/Editor';
+import { select } from './testUtils/selection';
+
+describe.each([
+    [
+        'bold',
+        'enabled',
+        'reversed',
+        '<strong>foo</strong>',
+        'fo<strong>o&nbsp;</strong>bar',
+        '<del>fo<strong>o</strong><strong><br/>b</strong>ar</del>',
+        '<strong>fo</strong>o <strong>bar</strong>',
+    ],
+    [
+        'italic',
+        'enabled',
+        'reversed',
+        '<em>foo</em>',
+        'fo<em>o&nbsp;</em>bar',
+        '<del>fo<em>o</em><em><br/>b</em>ar</del>',
+        '<em>fo</em>o <em>bar</em>',
+    ],
+    [
+        'underline',
+        'enabled',
+        'reversed',
+        '<u>foo</u>',
+        'fo<u>o&nbsp;</u>bar',
+        '<del>fo<u>o</u><u><br/>b</u>ar</del>',
+        '<u>fo</u>o <u>bar</u>',
+    ],
+    [
+        'strikethrough',
+        'enabled',
+        'reversed',
+        '<del>foo</del>',
+        'fo<del>o&nbsp;</del>bar',
+        '<del>fo</del>o<br/>b<del>ar</del>',
+        '<del>fo</del>o <del>bar</del>',
+    ],
+])(
+    'Formatting %s',
+    (
+        name,
+        defaultState,
+        expectedActivationState,
+        expectedSimpleFormatting,
+        expectedAdvancedFormatting,
+        expectedMultipleLineFormatting,
+        expectedUnformatting,
+    ) => {
+        let button: HTMLButtonElement;
+        let textbox: HTMLDivElement;
+
+        beforeEach(async () => {
+            render(<Editor />);
+            textbox = screen.getByRole('textbox');
+            await waitFor(() =>
+                expect(textbox).toHaveAttribute('contentEditable', 'true'),
+            );
+            button = screen.getByRole('button', { name });
+        });
+
+        it(`Should be ${defaultState} by default`, async () => {
+            // Then
+            expect(button).toHaveAttribute('data-state', defaultState);
+        });
+
+        it(
+            `Should be ${expectedActivationState} ` + `after single activation`,
```

Comment:
do we need to stop and restart the backquotes here?

Reply:
Yes, conflict with prettier and eslint maxLen.

Prettier doesn't format the it/test function in test suite but the eslint maxLen applied...
The line needs to be split manually

More information here https://github.com/prettier/eslint-config-prettier#max-len